### PR TITLE
refactor: selected option typing

### PR DESCRIPTION
**Pull request by MykhailoButkevych-MR**
1. Fix `as` ts definitions for selected option.
2. Add `selectedOptionResolver` helper function.
3. Remove `searchable` prop from `Select` childs, instead it can be verified based on `searchValue`.
4. `searchValue` can be null which means `Select` isn't searchable.
5. Change prop name `selectedOption` to `selectedOptions` for `MultiSelect`

### CODE REVIEW

File: packages/react-native-select-pro/src/components/options-list/options-list.tsx [hunk 1]
```diff
@@ -62,6 +60,9 @@ export const OptionsList = ({
         sectionHeaderTextStyle,
         containerStyle,
     } = optionsListStyles ?? {};
+    const optionsWithSections = isSectionOptionsType(optionsData);
```
Comment (kkaplinski) [inline packages/react-native-select-pro/src/components/options-list/options-list.tsx:63]: maybe: `isSectionOptionsData` or `isOptionsWithSections` , when we have boolean we should try to use words with `is`

File: packages/react-native-select-pro/src/helpers/selected-option-resolver.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { OptionType } from '../types';
```
Comment (irekrog) [inline packages/react-native-select-pro/src/helpers/selected-option-resolver.ts:1]: add to index.ts in helpers

File: packages/react-native-select-pro/src/components/multi-select/multi-select.tsx [hunk 3]
```diff
@@ -29,30 +28,45 @@ export const MultiSelect = ({
...
-                    placeholderTextColor={placeholderTextColor}
-                    textStyle={textStyle}
-                    multiSelectionOptionStyle={multiSelectionOptionStyle}
-                />
-            );
-        }
+    const resolveNoSelectedOptions = () => {
+        const selectInput = (
+            <SelectInput
+                disabled={disabled}
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/multi-select/multi-select.tsx:36]: {...disabled,...}

File: packages/react-native-select-pro/src/components/multi-select/multi-select.tsx [hunk 4]
```diff
@@ -29,30 +28,45 @@ export const MultiSelect = ({
...
+                textStyle={textStyle}
+                multiSelectionOptionStyle={multiSelectionOptionStyle}
+            />
+        );
+        return isSearchable ? selectInput : placeholderMultiOption;
+    };
 
-        const optionWidth = () => {
+    const resolveSelectedOptions = (selectedOptions: OptionType[]) => {
+        const calculateWidth = () => {
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/multi-select/multi-select.tsx:66]: you can create a custom hook eg useMultiSelect and with returned calculateWidth

File: packages/react-native-select-pro/src/components/multi-select/multi-select.tsx [hunk 5]
```diff
@@ -29,30 +28,45 @@ export const MultiSelect = ({
...
             const WIDTH_THRESHOLD = 100;
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/multi-select/multi-select.tsx:67]: constants outside function / component

File: packages/react-native-select-pro/src/components/multi-selected-option/multi-selected-option.tsx [hunk 6]
```diff
@@ -25,7 +24,7 @@ export const MultiSelectedOption = ({
                 multiSelectionOptionStyle,
                 { width: optionWidth },
             ]}
-            onPress={() => (onPressRemove ? onPressRemove(option as OptionType) : undefined)}
+            onPress={() => (onPressRemove && option ? onPressRemove(option) : undefined)}
```
Comment (irekrog) [inline packages/react-native-select-pro/src/components/multi-selected-option/multi-selected-option.tsx:27]: `onPress={onPressRemove && option ? () => onPressRemove(option) : undefined}`

**Review by irekrog — APPROVED** [review]

**Review by kkaplinski — APPROVED** [review]

**Pull request merged**